### PR TITLE
publiccloud: Do not add region for PUBLIC_CLOUD_SLES4SAP on destroy

### PR DESCRIPTION
**Pull request by cfconrad**
This is a follow up for https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/13289
and try to address the issue with SLES4SAP terraform profiles.

- Ticked: https://progress.opensuse.org/issues/63604
- Verification run: http://1b210.qa.suse.de/t9265

### CODE REVIEW

File: lib/publiccloud/provider.pm [hunk 1]
```diff
@@ -441,18 +441,21 @@ Destroys the current terraform deployment
 =cut
 sub terraform_destroy {
     my ($self) = @_;
+    my $cmd;
```
Comment (asmorodskyi) [inline lib/publiccloud/provider.pm:444]:
```suggestion
    my $cmd = 'terraform destroy -no-color -auto-approve';
```

**Comment by asmorodskyi** [comment]
but this would basically means that `destroy` will be still broken for SAP/HA because as @cfconrad  discovered in case of AWS region is *required* for destroy call ?

**Comment by cfconrad** [comment]
They are using a tfvars file. And in that file the region is specified. 
From https://www.terraform.io/docs/language/values/variables.html#variable-definitions-tfvars-files you can see that a file which is called `terraform.tfvars` get's loaded automatically.

So I think it should work!

File: lib/publiccloud/provider.pm [hunk 2]
```diff
@@ -441,18 +441,21 @@ Destroys the current terraform deployment
...
     else {
         assert_script_run('cd ' . TERRAFORM_DIR);
+        # Add region variable also to `terraform destroy` (poo#63604) -- needed by AWS.
+        $cmd = sprintf(q(terraform destroy -no-color -auto-approve -var 'region=%s'), $self->region);
     }
     # Retry 3 times with considerable delay. This has been introduced due to poo#95932 (RetryableError)
     # terraform keeps track of the allocated and destroyed resources, so its safe to run this multiple times.
-    # Add region variable also to `terraform destroy` (poo#63604) -- needed by AWS.
-    my $ret = script_retry(sprintf(q(terraform destroy -no-color -auto-approve -var 'region=%s'), $self->region), retry => 3, delay => 60, timeout => get_var('TERRAFORM_TIMEOUT', TERRAFORM_TIMEOUT), die => 0);
+    my $ret = script_retry($cmd, delay => 60, timeout => get_var('TERRAFORM_TIMEOUT', TERRAFORM_TIMEOUT), die => 0);
```
Comment (ldevulder) [inline lib/publiccloud/provider.pm:458]: The `retry => 3` option has been removed, is it expected? If so then the comment above about this retry should also be removed.
If this is not expected then I will re-add this option in PR #13338.

Comment (asmorodskyi) [reply lib/publiccloud/provider.pm:458]: nice catch , yeah I don't see good reason why to remove it , so please add it where you suggested

Comment (cfconrad) [reply lib/publiccloud/provider.pm:458]: nice catch, thx

**Pull request merged**